src/Compatibility/Maps/src/Android/MapRenderer.cs

```diff
@@ -158,7 +157,7 @@ protected override void OnElementChanged(ElementChangedEventArgs<Map> e)
 
 			Control.GetMapAsync(this);
 
-			MessagingCenter.Subscribe<Map, MapSpan>(this, MoveMessageName, OnMoveToRegionMessage, Map);
+			WeakReferenceMessenger.Default.Register<MapRenderer, MapSpan>(this, (renderer, args) => OnMoveToRegionMessage(renderer.Element, args));
```

Comment:
This is introducing a closure on the recipient:

```suggestion
			WeakReferenceMessenger.Default.Register<MapRenderer, MapSpan>(this, (r, args) => r.OnMoveToRegionMessage(r.Element, args));
```